src/test/scala/org/broadinstitute/dsde/firecloud/service/TSVFileSupportSpec.scala

```diff
@@ -58,6 +58,48 @@ class TSVFileSupportSpec extends AnyFreeSpec with TSVFileSupport {
     }
   }
 
+  "stringToTypedAttribute" - {
+    val booleanTestCases = Map("true" -> AttributeBoolean(true), "True" -> AttributeBoolean(true), "tRue" -> AttributeBoolean(true),
+      "false" -> AttributeBoolean(false), "False" -> AttributeBoolean(false), "FALSE" -> AttributeBoolean(false)
+    )
+    val integerTestCases = Map(
+      "525600" -> AttributeNumber(525600), "525,600" -> AttributeString("525,600"), "525_600" -> AttributeString("525_600"),
+      "-525600" -> AttributeNumber(-525600), "-525,600" -> AttributeString("-525,600"), "-525_600" -> AttributeString("-525_600"),
+      "0" -> AttributeNumber(0), "00" -> AttributeNumber(0), "-0" -> AttributeNumber(0), "-00" -> AttributeNumber(0)
+    )
+    val doubleTestCases = Map(
+      "4.2" -> AttributeNumber(4.2), ".42" -> AttributeNumber(0.42), "0.42" -> AttributeNumber(0.42),
+      "42." -> AttributeNumber(42), "42.0" -> AttributeNumber(42), "." -> AttributeString("."),
+      "-4.2" -> AttributeNumber(-4.2), "-.42" -> AttributeNumber(-0.42), "-0.42" -> AttributeNumber(-0.42),
+      "-42." -> AttributeNumber(-42), "-42.0" -> AttributeNumber(-42), "-." -> AttributeString("-."),
+    )
+    val stringTestCases = List("", "string", "true525600", ",")
+
+    booleanTestCases foreach {
+      case (input, expected) => s"boolean test: $input" in {
+        stringToTypedAttribute(input) shouldBe expected
+      }
+    }
+
+    integerTestCases foreach {
+      case (input, expected) => s"int test: $input" in {
+        stringToTypedAttribute(input) shouldBe expected
+      }
+    }
+
+    doubleTestCases foreach {
+      case (input, expected) => s"double test: $input" in {
+        stringToTypedAttribute(input) shouldBe expected
+      }
+    }
```

Comment:
I know this was my suggestion during pairing, but in practice having one test case for each input MAY be overwhelming. You end up with test output like:
```
stringToTypedAttribute
[info] - boolean test: false
[info] - boolean test: yes
[info] - boolean test: True
[info] - boolean test: False
[info] - boolean test: FALSE
[info] - boolean test: 1
[info] - boolean test:   true 
[info] - boolean test: tRue
[info] - boolean test: 0
[info] - boolean test: true
[info] - boolean test: no
[info] - int test: -525,600
[info] - int test: -0
[info] - int test: 525,600
[info] - int test: -525600
[info] - int test: 525_600
[info] - int test: 525600
[info] - int test: -00
[info] - int test: 0
```
you could instead make a single test case that loops through the different input values, and use `withClue` to help with any failures. This would look something like:
```scala
    "should detect double values" in {
      doubleTestCases foreach {
        case (input, expected) => withClue(s"for input value '$input'") {
          stringToTypedAttribute(input) shouldBe expected
        }
      }
    }
``` 
and would give you shorter output, like:
```
[info] stringToTypedAttribute
[info] - should detect boolean values
[info] - should detect integer values
[info] - should detect double values *** FAILED ***
[info]   for input value '4.9E-324' AttributeNumber(4.9E-324) was not equal to AttributeNumber(-1.7976931348623157E+308) (TSVFileSupportSpec.scala:111)
```
( I created an intentionally failing test to illustrate the output of `withClue` ).

The change in behavior is that the latter approach will fail the test on the first failed assertion, whereas the former approach will always test all input values, potentially resulting in multiple failures. I could go either way but am leaning towards the latter.

Finally - by convention, Scalatest test names start/involve the word "should", so even if you kept with the former approach, you'd want something like `case (input, expected) => s"should handle ints for input: $input" in {` instead of `case (input, expected) => s"int test: $input" in {`.